### PR TITLE
[NFC] Extract function to get component permissions

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Function extraction as part of code simplification

Before
----------------------------------------
Code harder to read

After
----------------------------------------
Code easier to read

Technical Details
----------------------------------------
Just an extraction

Comments
----------------------------------------
I will follow this with fixes described in https://lab.civicrm.org/dev/core/issues/454 with the end goal of making getcount work with permissions

### CODE REVIEW

File: CRM/Activity/BAO/Activity.php [hunk 1]
```diff
@@ -2772,7 +2730,7 @@ public static function checkPermission($activityId, $action) {
     $targetID = CRM_Utils_Array::key('Activity Targets', $activityContacts);
 
     // Check for source contact.
-    if (!$componentId || $allow) {
```
Comment (seamuslee001) [inline CRM/Activity/BAO/Activity.php]: @eileenmcnaughton this seems to be a bit odd given that what if there is no component id?

Comment (eileenmcnaughton) [reply CRM/Activity/BAO/Activity.php]: @seamuslee001 yeah I remove componentId there because $componentId is only ever true when allow is also true

**Comment by seamuslee001** [comment]
I am satisfied that nothing is changing here and it is in deed NFC

**Comment by seamuslee001** [comment]
Merging

**Comment by eileenmcnaughton** [comment]
thanks @seamuslee001 !

**Pull request merged**